app/frontend/components/common/asset.tsx

```diff
@@ -74,13 +74,11 @@ export const FormattedAssetItem = ({
     if (type === AssetFamily.ADA) return <span className={styles.assetName}>ADA</span>
     return (
       <span>
-        {assetName ? (
-          <span className={styles.assetName}>{assetNameHex2Readable(assetName)}</span>
-        ) : (
-          <span className={styles.empty}>
-            {'<'}no name{'>'}
+        {
+          <span className={metadata || assetName ? styles.assetName : styles.empty}>
+            {metadata?.name ?? assetNameHex2Readable(assetName) ?? '<no name>'}
```

Comment:
This would work incorrectly if the token has no metadata and an empty asset name hex - we would just show an empty string whereas before this PR this got formatted to `<no name>` which makes more sense.

I would suggest refactorng this logic into a pure helper function `formatAssetName()` which would explicitly receive the token assetNameHex and metadata (if present) and have explicit conditions that are easy to follow as I personally got a bit lost when evaluating this composite condition at first and it can hide pretty nasty edge-cases as the one I pointed out

Reply:
Fixed